### DISCUSSION TITLE
AWS Elastic Beanstalk Flask vs FastAPI memory usage

### DISCUSSION THREAD

**Discussion by andrew-lyons** [fastapi/fastapi#3422] [Questions]
Hi! I've been hunting down a high memory issue with AWS EB that only appeared when we started testing FastAPI. We've been loving the performance boost it's given us over Flask, but we noticed some weird behavior when we import Sagemaker.

At one point we used the sagemaker sdk, and then moved to other means--but the import remained. So in Flask, we have an `import sagemaker`, but don't actually use it. We then deploy this to our beanstalk instance and monitor the health to find no issues. All health signs are good and we have very low memory/cpu usage at no request load.

When we tried moving the code over to a FastAPI elastic beanstalk server, we reached ~95% memory usage give or take a few %. Eventually I found that we still had the needless `import sagemaker` and removed it. When I deployed to EB, it solved the memory issue, and we went down to healthy levels. Even with high memory usage it was much faster, but we incur more charges from AWS due to the large memory usage at idle.

It's possible I missed something, but the issue was only fixed by removing the import--one line of code. Is there something inherent to FastAPI's operation that would cause the spike in memory usage from the import over Flask? We checked both beanstalk instances and there is no difference in memory allocation/scaling/etc.

**Comment by YuriiMotov** [comment]
It's hard to say anything without some code
